code/modules/antagonists/dodgeball/dodgeball.dm

```diff
@@ -0,0 +1,92 @@
+/datum/antagonist/dodgeball
+	name = "Doomballer"
+	var/obj/item/toy/beach_ball/doomball/mine
+	show_in_antagpanel = FALSE
+	show_name_in_check_antagonists = TRUE
+	can_hijack = HIJACK_HIJACKER
+
+/datum/antagonist/dodgeball/apply_innate_effects(mob/living/mob_override)
+	var/mob/living/L = owner.current || mob_override
+	ADD_TRAIT(L, TRAIT_NOGUNS, "doomballer")
+
+/datum/antagonist/dodgeball/remove_innate_effects(mob/living/mob_override)
+	var/mob/living/L = owner.current || mob_override
+	REMOVE_TRAIT(L, TRAIT_NOGUNS, "doomballer")
+
+/datum/antagonist/dodgeball/proc/forge_objectives()
+	var/datum/objective/steal/steal_objective = new
+	steal_objective.owner = owner
+	objectives += steal_objective
+
+	var/datum/objective/hijack/hijack_objective = new
+	hijack_objective.explanation_text = "Escape on the shuttle alone. Ensure that nobody else makes it out."
+	hijack_objective.owner = owner
+	objectives += hijack_objective
+
+	owner.objectives |= objectives
+
+/datum/antagonist/dodgeball/on_gain()
+	forge_objectives()
+	owner.special_role = "doomballer"
+	give_equipment()
+	. = ..()
+
+/datum/antagonist/dodgeball/greet()
+	to_chat(owner, "<span class='boldannounce'>Your doomball cries out for souls. Claim the lives of others, or die trying</span>")
+
+	owner.announce_objectives()
+
+/datum/antagonist/dodgeball/proc/give_equipment()
+	var/mob/living/carbon/human/H = owner.current
+	if(!istype(H))
+		return
+
+	for(var/obj/item/I in H.get_equipped_items(TRUE))
+		qdel(I)
+	for(var/obj/item/I in H.held_items)
+		qdel(I)
+	H.equip_to_slot_or_del(new /obj/item/clothing/under/kilt/highlander(H), SLOT_W_UNIFORM)
+	H.equip_to_slot_or_del(new /obj/item/radio/headset/heads/captain(H), SLOT_EARS)
+	H.equip_to_slot_or_del(new /obj/item/clothing/head/beret/highlander(H), SLOT_HEAD)
+	H.equip_to_slot_or_del(new /obj/item/clothing/shoes/combat(H), SLOT_SHOES)
+	H.equip_to_slot_or_del(new /obj/item/pinpointer/nuke(H), SLOT_L_STORE)
```

Comment:
honestly, this stuff should part of an outfit datum, but I guess you can't be troubled by improving this rather than copypasting.

Reply:
I don't understand byondcode well enough to confidently do much more than copypasting. I did the dating sim, but that was using the html shit. I am very tired, so just ignore me I guess. lol

Reply:
Yea, take it easy. Learn a bit. Dream maker code is dummy level